datafusion/common/src/stats.rs

```diff
@@ -17,11 +17,184 @@
 
 //! This module provides data structures to represent statistics
 
-use std::fmt::Display;
-
+use crate::ScalarValue;
 use arrow::datatypes::DataType;
 
-use crate::ScalarValue;
+use std::fmt::{self, Debug, Display};
+
+/// To deal with information without exactness guarantees, we wrap it inside a
+/// [`Sharpness`] object to express its reliability.
```

Comment:
```suggestion
/// Represents a value with a degree of certainty. `Sharpness` is used to 
/// propagate information the precision of statistical values.
```

While writing this I wonder if naming `Sharpness` to  `Precision` might match more closely with what this type represents 🤔